cmd/geranos/walkmanifests.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+)
+
+type walkManifestsFunc func(ref name.Reference) error
+
+// Unfortunately this is only doable on GCP currently.
+//
+// TODO: To support other registries in the meantime, we could require a list of
```

Comment:
You could use the promotion manifests, I guess

Reply:
For Kubernetes I'm not worried about this.

I'd like the tool to be reusable by other projects, and currently that requires the use of GCR/AR (or using their own tool to populate the blob storage while reusing archeio), but that is low-priority to fix versus getting ourselves off of rclone.